src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -137,3 +147,97 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
         if comment:
             query += f" comment='{comment}'"
         return self._execute_query(query)
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        parameters: Optional[List[str]] = None,
+    ):
+        files_list = self._get_files_list_from_stage(stage_path)
+        filtered_list = self._filter_files_list(stage_path, files_list)
+        sorted_list = sorted(
+            filtered_list, key=cmp_to_key(self._stage_files_comparator)
+        )
+
+        sql_parameters = self._parse_execute_parameters(parameters)
+        results = []
+        for file in sorted_list:
+            results.append(self._call_execute_immediate(file, sql_parameters, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = stage_path.split("/")[0]
+        files_list_result = self.list_files(stage_name).fetchall()
```

Comment:
Weren't we supposed to use `iter_stage` from #942?

Reply:
I hoped so, but it was easier to filter files as list and you wanted to execute them in order, so I need whole list to sort it.

I can use `iter_stage` filter files one by one, collect and sort, but filtering will not be that efficient like in current way.

WDYT?